app/helpers/basespace_helper.rb

```diff
@@ -19,14 +19,18 @@ def revoke_access_token(access_token)
 
   def verify_access_token_revoked(access_token, sample_id)
     # Verify that the token was revoked by using it to call an API endpoint.
-    # The API endpoint should return a 401.
 
     fetch_from_basespace(BASESPACE_CURRENT_PROJECTS_URL, access_token, {}, true)
 
+    # If we reach this step, the access token must not have been revoked.
```

Comment:
Hey @davidrissato, I realize you had commented that this function feels counterintuitive. I believe this is a rare usage of the fetch_from_basespace http method, and almost all of the time, we won't expect the request to fail, so I'm okay with the function being a little unusual (I added a little more commenting). I'd rather not modify the http method to make it more complicated at this point, unless these cases become more common.

I did add a custom HttpError type that stores the status code. This ensures that we only catch the one specific error we are expecting and re-raise all other errors.